enterprise/utils.py

```diff
@@ -2306,3 +2308,23 @@ def get_default_invite_key_expiration_date():
     The default expiration date is 365 days from the current date.
     """
     return localized_utcnow() + datetime.timedelta(days=365)
+
+
+def ensure_course_enrollment_is_allowed(course_id, email, enrollment_api_client):
+    """
+    Creates a CourseEnrollmentAllowed object for initiation only courses.
+
+    Arguments:
+        course_id (str): ID of the course to allow enrollment
+        email (str): email of the user whose enrollment should be allowed
+        enrollment_api_client (:class:`enterprise.api_client.lms.EnrollmentApiClient`): Enrollment API Client
```

Comment:
Tiny nit: wouldn't it be better to add the typing directly to the arguments (line 2313)?

Reply:
@tecoholic, `typing` is a relatively recent feature in Python, so most people tend to forget about it. While following the convention is important, we should also be open to introducing apparent improvements. We are already gradually adding the `typing` to more and more repositories in Open edX. Examples: https://github.com/openedx/edx-platform/pull/32591, https://github.com/openedx/opaque-keys/pull/256.

This is not a blocker here, but something to be aware of.

Reply:
@Agrendalath Alright, in that case, let this PR be the one to bring in the change. I have added the typing for the utility function introduced in this PR.